src/shared/mount-util.c

```diff
@@ -37,55 +34,44 @@ int umount_recursive(const char *prefix, int flags) {
          * unmounting them until they are gone. */
 
         do {
-                _cleanup_fclose_ FILE *proc_self_mountinfo = NULL;
+                _cleanup_(mnt_free_tablep) struct libmnt_table *table = NULL;
+                _cleanup_(mnt_free_iterp) struct libmnt_iter *iter = NULL;
 
                 again = false;
-                r = 0;
 
-                proc_self_mountinfo = fopen("/proc/self/mountinfo", "re");
-                if (!proc_self_mountinfo)
-                        return -errno;
+                table = mnt_new_table();
+                iter = mnt_new_iter(MNT_ITER_FORWARD);
+                if (!table || !iter)
+                        return log_oom();
 
```

Comment:
still logs at non-debug level

Reply:
This one is different, it was logging at non-debug level before, I just kept the behaviour.

Reply:
but please. let's fix this when touching that

Reply:
It seems you were right, and I added logging here, but then removed it in one of the later commits. I now fixed this during the rebase to have `return -ENOMEM`.